Framework/API/src/AlgorithmFactory.cpp

```diff
@@ -44,35 +44,47 @@ std::shared_ptr<Algorithm>
 AlgorithmFactoryImpl::create(const std::string &name,
                              const int &version) const {
   int local_version = version;
-  if (version < 0) {
-    if (version == -1) // get latest version since not supplied
-    {
-      auto it = m_vmap.find(name);
-      if (!name.empty()) {
-        if (it == m_vmap.end())
-          throw std::runtime_error("Algorithm not registered " + name);
-        else
-          local_version = it->second;
-      } else
-        throw std::runtime_error(
-            "Algorithm not registered (empty algorithm name)");
+  // Version not supplied
+  if (version == -1) {
+    auto v_it = m_vmap.find(name);
+    if (v_it == m_vmap.end()) {
+      // Try finding version from alias instead
+      auto a_it = m_amap.find(name);
+      if (a_it != m_amap.end()) {
+        v_it = m_vmap.find(a_it->second);
+      }
     }
+    if (v_it == m_vmap.end()) // Version could not be found
+      throw std::runtime_error("Algorithm not registered " + name);
+    else
+      local_version = v_it->second;
   }
+
   try {
     return this->createAlgorithm(name, local_version);
   } catch (Kernel::Exception::NotFoundError &) {
-    auto it = m_vmap.find(name);
-    if (it == m_vmap.end())
-      throw std::runtime_error("algorithm not registered " + name);
-    else {
-      g_log.error() << "algorithm " << name << " version " << version
-                    << " is not registered \n";
-      g_log.error() << "the latest registered version is " << it->second
-                    << '\n';
-      throw std::runtime_error("algorithm not registered " +
-                               createName(name, local_version));
+  }
+  auto v_it = m_vmap.find(name);
```

Comment:
I see this is what it did before but is there a need to search the version map again here? Do `version` & `local_version` not contain the information we want already?